src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1060,6 +1063,23 @@ CreateDistributedPlan(uint64 planId, Query *originalQuery, Query *query, ParamLi
 	 */
 	if (list_length(subPlanList) > 0 || hasCtes)
 	{
+		/*
+		 * recursive planner should handle all the tree from bottom to
+		 * top at single pass. i.e. It should have already recursively planned all
+		 * required parts in its first pass. Hence, we expect previousSubPlanList
+		 * to be empty. Otherwise, this means we have bug at recursive planner,
+		 * which needs to be handled. We add a check here and return error.
+		 */
+		if (list_length(previousSubPlanList) > 0)
+		{
+			const char *multiPassRecursivePlanErrorMessage =
```

Comment:
I cannot easily follow the term `multiPassRecursivePlanErrorMessage`. Do we even need the variable at all?

Also, why do we have both DEBUG1 and ERROR? The former seems not that useful?

Reply:
We fallback to cte planning when inlining fails. In that case, ERROR message is swallowed and prints as DEBUG4.

Reply:
I still think that ERROR is enough. If cte inlining fails, and you want to learn why, you should already have DEBUG4 enabled.